### PR TITLE
Avatar: Add size 16

### LINKED ISSUE #22977: [Feature]: Add size 16 variant to Avatar component
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

Would like to add an additional size variant (size 16) to the Fluent 2 Avatar component. 

### Have you discussed this feature with our team

Paul Gildea, Ben Howell

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by behowell**
## New Behavior

The designs for Avatar have been updated to include size 16. This implements the new size.

## Related Issue(s)

* Fixes #22977

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                          |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr> | `45.314 kB`<br />`13.175 kB` | `45.502 kB`<br />`13.255 kB` | `188 B` <br />`80 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `179.813 kB`<br />`50.19 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `34.178 kB`<br />`11.138 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/47008296a95fce3467a0fa94c809155afb621d87'>47008296a95fce3467a0fa94c809155afb621d87</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/Avatar-mount_base.html">1140</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/Avatar-mount.html">1085</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/Button-mount_base.html">691</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/Button-mount.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProvider-mount_base.html">2329</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProvider-mount.html">2288</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-mount_base.html">341</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-mount.html">386</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">313</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">363</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">365</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/MakeStyles-mount_base.html">1926</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23003/perf-test-react-components/MakeStyles-mount.html">1951</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by behowell** [comment]
I reviewed and accepted the new screener states that include size 16.

**Review by spmonahan — APPROVED** [review]

**Comment by behowell** [comment]
@spmonahan Re-requesting your review since I made some more changes to getInitials after you approved. Thanks!

File: packages/react-components/react-avatar/src/utils/getInitials.ts [hunk 1]
```diff
@@ -34,21 +34,22 @@ const MULTIPLE_WHITESPACES_REGEX: RegExp = /\s+/g;
 // eslint-disable-next-line @fluentui/max-len
 const UNSUPPORTED_TEXT_REGEX: RegExp = /[\u0600-\u06FF\u0750-\u077F\u08A0-\u08FF\u1100-\u11FF\u3130-\u318F\uA960-\uA97F\uAC00-\uD7AF\uD7B0-\uD7FF\u3040-\u309F\u30A0-\u30FF\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF]|[\uD840-\uD869][\uDC00-\uDED6]/;
 
-function getInitialsLatin(displayName: string, isRtl: boolean): string {
+function getInitialsLatin(displayName: string, isRtl: boolean, firstInitialOnly?: boolean): string {
   let initials = '';
 
   const splits: string[] = displayName.split(' ');
```
Comment (spmonahan) [inline packages/react-components/react-avatar/src/utils/getInitials.ts:40]: This is definitely an edge case and I'm not sure how much we want to handle bad input but there are ways in which splitting this way can cause failures when subsequently building up the initials.

```js
// Three spaces between the words
"First   Last".split(" "); // ["First", "", "", "Last"]
```

**Review by spmonahan — APPROVED** [review]

**Pull request merged**